nemo_text_processing/inverse_text_normalization/vi/taggers/date.py

```diff
@@ -19,143 +19,143 @@
+        )
+        graph_thousands = (
             graph_digit
             + delete_space
             + pynutil.delete(pynini.union("nghìn", "ngàn"))
             + delete_space
             + graph_hundred_component
             + delete_space
-            + (graph_teen | graph_ties | graph_digits)
+            + pynini.union(graph_teen, graph_ties, graph_digits)
         )
-        return graph
-
-    graph_ties = _get_ties_graph()
-    graph_digits = _get_digits_graph()
-    graph_hundreds = _get_hundreds_graph(graph_ties, graph_digits)
-    graph_thousands = _get_thousands_graph(graph_ties, graph_digits)
-    year_graph = (
-        # 20 19, 40 12, 2012, 2 0 0 5, 2 0 17, 938 - assuming no limit on the year
-        graph_digit
-        + delete_space
-        + (graph_digit | graph_zero)
-        + delete_space
-        + (graph_teen | graph_ties | graph_digits)
-        | graph_thousands
-        | graph_hundreds
-        | (graph_digit + pynutil.insert("0") + delete_space + (graph_ties | graph_digits | graph_teen))
-    )
-    year_graph.optimize()
-    return year_graph
-
-
-class DateFst(GraphFst):
-    """
-    Finite state transducer for classifying date,
-        e.g. mười lăm tháng một năm hai nghìn mười hai -> date { day: "15" month: "1" year: "2012" preserve_order: true }
-        e.g. ngày ba mốt tháng mười hai năm một chín chín chín -> date { day: "31" month: "12" year: "2012" preserve_order: true }
-        e.g. năm hai không hai mốt -> date { year: "2021" preserve_order: true }
-
-    Args:
-        cardinal: CardinalFst
-    """
-
-    def __init__(self, cardinal: GraphFst):
-        super().__init__(name="date", kind="classify")
-
-        cardinal_graph = cardinal.graph_no_exception
-        year_graph = _get_year_graph()
-        YEAR_WEIGHT = 0.001
-        year_graph = pynutil.add_weight(year_graph, YEAR_WEIGHT)
-        month_graph = _get_month_graph()
 
-        month_graph = pynutil.insert('month: "') + month_graph + pynutil.insert('"')
+        # Complete year graph with all supported patterns
+        year_graph_raw = pynini.union(
+            # Standard patterns: 2019, 2012, 2005, etc.
+            graph_digit
+            + delete_space
+            + pynini.union(graph_digit, graph_zero)
+            + delete_space
+            + pynini.union(graph_teen, graph_ties, graph_digits),
+            graph_thousands,
+            graph_hundreds,
+            (graph_digit + pynutil.insert("0") + delete_space + pynini.union(graph_ties, graph_digits, graph_teen)),
+            (
+                pynini.union(graph_digit, graph_zero)
+                + delete_space
+                + pynini.union(graph_digit, graph_zero)
+                + delete_space
+                + pynini.union(graph_digit, graph_zero)
+                + delete_space
+                + pynini.union(graph_digit, graph_zero)
+            ),
+        ).optimize()
+
+        year_graph = pynutil.add_weight(year_graph_raw, YEAR_WEIGHT)
```

Comment:
do you need a dedicated weight or can you just reuse a general one ("min weight) for instance)